### ISSUE TITLE
[BUG]: Could not load Custom Materializer class when using local Kubeflow stack

### ISSUE THREAD

**Issue by Jelena811** [zenml-io/zenml#479]
### Contact Details [Optional]

tosic.jelena811@gmail.com penenadpi@gmail.com

### What happened?

An error occurred when trying to run example _custom_materializer_ from provided _zenml_examples_ repo, using local_kubeflow_stack (as described [here](https://docs.zenml.io/guides/functional-api/deploy-to-production)). 
An error looks like this
<img width="811" alt="Screenshot 2022-03-25 at 10 07 01" src="https://user-images.githubusercontent.com/59333923/160092843-ad87e075-53d4-46bf-8f00-bde377051a5a.png">

And it seems to be caused by wrong class name parsing, which can be seen here:
<img width="619" alt="Screenshot 2022-03-25 at 10 07 51" src="https://user-images.githubusercontent.com/59333923/160093019-370e6060-2ef3-4785-afb8-dc773aa4c018.png">

As it can be seen, the first character has been omitted and that's why the module cannot be loaded. The program is trying to load _un.MyMaterializer_ and not _run.Materializer_ (as that is the name of python file we are running in this example)

### Reproduction steps

1. zenml  example pull custom_materializer
2. cd zenml_examples
3. cd custom_materializer
4. zenml init
5. zenml container-registry register local_registry --type=default --uri=localhost:5000 
6. zenml orchestrator register kubeflow_orchestrator --type=kubeflow
7. zenml stack register local_kubeflow_stack \
    -m local_metadata_store \
    -a local_artifact_store \
    -o kubeflow_orchestrator \
    -c local_registry

8. zenml stack set local_kubeflow_stack
9. zenml stack up
10. python3 run.py
...

### ZenML Version

0.6.2

### Python Version

3.8

### OS Type

Mac OS

### Relevant log output

```shell
47
│ /usr/local/lib/python3.8/dist-packages/zenml/utils/source_utils.py:314 in    │
2648
│ import_class_by_path                                                         │
2649
│                                                                              │
2650
│   311 │   """                                                                │
2651
│   312 │   classname = class_path.split(".")[-1]                              │
2652
│   313 │   modulename = ".".join(class_path.split(".")[0:-1])                 │
2653
│ ❱ 314 │   mod = importlib.import_module(modulename)                          │
2654
│   315 │   return getattr(mod, classname)  # type: ignore[no-any-return]      │
2655
│   316                                                                        │
2656
│   317                                                                        │
2657
│                                                                              │
2658
│ ╭───────────── locals ─────────────╮                                         │
2659
│ │ class_path = 'un.MyMaterializer' │                                         │
2660
│ │  classname = 'MyMaterializer'    │                                         │
2661
│ │ modulename = 'un'                │                                         │
2662
│ ╰──────────────────────────────────╯                                         │
2663
│                                                                              │
2664
│ /usr/lib/python3.8/importlib/__init__.py:127 in import_module                │
2665
│                                                                              │
2666
│   124 │   │   │   if character != '.':                                       │
2667
│   125 │   │   │   │   break                                                  │
2668
│   126 │   │   │   level += 1                                                 │
2669
│ ❱ 127 │   return _bootstrap._gcd_import(name[level:], package, level)        │
2670
│   128                                                                        │
2671
│   129                                                                        │
2672
│   130 _RELOADING = {}                                                        │
2673
│                                                                              │
2674
│ ╭──── locals ────╮                                                           │
2675
│ │   level = 0    │                                                           │
2676
│ │    name = 'un' │                                                           │
2677
│ │ package = None │                                                           │
2678
│ ╰────────────────╯                                                           │
2679
│ <frozen importlib._bootstrap>:1014 in _gcd_import                            │
2680
│ ╭──── locals ────╮                                                           │
2681
│ │   level = 0    │                                                           │
2682
│ │    name = 'un' │                                                           │
2683
│ │ package = None │                                                           │
2684
│ ╰────────────────╯                                                           │
2685
│ <frozen importlib._bootstrap>:991 in _find_and_load                          │
2686
│ ╭────────────────────── locals ──────────────────────╮                       │
2687
│ │ import_ = <function _gcd_import at 0x7fd873815430> │                       │
2688
│ │  module = <object object at 0x7fd8737eb060>        │                       │
2689
│ │    name = 'un'                                     │                       │
2690
│ ╰────────────────────────────────────────────────────╯                       │
2691
│ <frozen importlib._bootstrap>:973 in _find_and_load_unlocked                 │
2692
│ ╭────────────────────── locals ──────────────────────╮                       │
2693
│ │ import_ = <function _gcd_import at 0x7fd873815430> │                       │
2694
│ │    name = 'un'                                     │                       │
2695
│ │  parent = ''                                       │                       │
2696
│ │    path = None                                     │                       │
2697
│ │    spec = None                                     │                       │
2698
│ ╰────────────────────────────────────────────────────╯                       │
2699
╰──────────────────────────────────────────────────────────────────────────────╯
2700
ModuleNotFoundError: No module named 'un'
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by htahir1** [comment]
This is a strange one. I am unsure why it would omit the character f. Could you share the code you used? I cannot replicate this at my end yet

**Comment by LXOXOL** [comment]
My name is on your list somehow. I am not in your company not do I wish to be.
I am asking you to look into this matter for me please. I believe I have been hacked by a woman in Chatham IL. I want you be aware of the fact that I had contacted the FBI and they are looking into it for me. I don't want you to be involved in this so please take care of this matter as soon as you can. She uses my Emails and also uses an Ipad staying that it's her primary tablet and my fire is what she uses occasionally. I am looking at a big headache. Please be advised that I am making you aware of the situation before you get involved too. I am reviewing a number of emails from I suppose are some of your clients. Please take my name and emails off of your company stuff. Thank you! L Mitchell.
Sent from Yahoo Mail on Android 
 
  On Mon, Mar 28, 2022 at 2:20 PM, Hamza ***@***.***> wrote:   


This is a strange one. I am unsure why it would omit the character f. Could you share the code you used? I cannot replicate this at my end yet

**Comment by htahir1** [comment]
@LXOXOL I do not know what this last message means, this is an open-source repository and your email is hidden.

@Jelena811 Could you kindly update this issue with your code? We can take it from there

**Comment by stefannica** [comment]
@Jelena811 It would be really useful to know the following:

* your operating system type (e.g. linux, macos or windows)
* the full path of your .zen folder
* the exact command you are using to run the custom_materializer example

**Comment by stefannica** [comment]
I re-tested this with the latest (0.7.2) and the bug seems to have been fixed, even earlier, in 0.7.1, when the ZenML profiles feature was introduced. There is still the slight possibility that the code indicated above doesn't work well in some corner-cases. but this is no longer a problem.
